core/src/CorePlatformInformationBase.cs

```diff
@@ -36,7 +36,11 @@ internal abstract class CorePlatformInformationBase
 
         public const string DefaultRedirectUri = "urn:ietf:wg:oauth:2.0:oob";
         public abstract string GetProductName();
-        public virtual Task<string> GetUserPrincipalNameAsync() { return null; }
```

Comment:
This causes a null ref exception because it returns a null Task instead of a null username.